.github/workflows/main.yml

```diff
@@ -20,14 +20,26 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
+        include:
+          - os: ubuntu-latest
+            optional_args: "-pt"
+          - os: windows-latest
+            optional_args: "-pt --cmake_gen ninja"
+          - os: macOS-latest
+            optional_args: ""
     steps:
-    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+    - name: Checkout sources
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+    - name: Setup Fortran Compiler
+      uses: fortran-lang/setup-fortran@8821f57b53846d35d62632eb51c60ac6c4bff4ce # v1.6.1
       with:
-        submodules: recursive
+          compiler: intel
     - name: Build C library
-      # TODO: Ubuntu is phasing out support for 32-bit packages (e.g., `apt install gcc-multilib`
-      # fails on GitHub's runner); only build the 64-bit version for now.
-      run: python buildall.py --force_bits 64
+      # 1. Force to only build the 64-bit version since ITT API 32-bit support will be discontinued soon
+      # 2. Disable PT support for MacOS since we have x86 specific assembly instruction
+      # 3. Switch to use Ninja CMake Generator for Windows since setup-fortran action
+      #    doesn't work in case of CMake + VS (https://github.com/fortran-lang/setup-fortran/issues/45)
```

Comment:
Good documentation! I can foresee people stumbling on this at some point.

Reply:
I hope this isn't trolling 😀

Reply:
No, no! I was just reading through this PR thinking, "oof, I hope I don't have to mess with this Fortran stuff in CI because I won't remember what works and what doesn't'! These doc comments came along at the right time.